server/src/main/java/org/elasticsearch/common/filesystem/LinuxFileSystemNatives.java

```diff
@@ -41,12 +41,21 @@
      * To allow the `struct stat' structure bits to vary without changing shared library major version number, the `stat' function is often
      * an inline wrapper around `xstat' which takes a leading version-number argument designating the data structure and bits used.
      *
-     * In glibc this version is defined in bits/stat.h (or bits/struct_stat.h in glibc 2.33) as:
+     * In glibc this version is defined in bits/stat.h (or bits/struct_stat.h in glibc 2.33, or bits/xstatver.h in more recent versions).
+     *
+     * For x86-64 the _STAT_VER used is:
      *  # define _STAT_VER_LINUX    1
      *  # define _STAT_VER _STAT_VER_LINUX
+     *
+     * For other architectures the _STAT_VER used is:
+     *  # define _STAT_VER_LINUX    0
+     *  # define _STAT_VER _STAT_VER_LINUX
```

Comment:
We really need https://github.com/elastic/elasticsearch/issues/80903 to deal with this sort of nonsense.

Reply:
I agree, I'm a bit afraid if the _xstat function changes over time and we don't catch it before in our tests.